integrations/qdrant/src/haystack_integrations/document_stores/qdrant/document_store.py

```diff
@@ -407,24 +490,59 @@ def _set_up_collection(
-        self.client.recreate_collection(
-            collection_name=collection_name,
-            vectors_config=rest.VectorParams(
-                size=embedding_dim,
-                on_disk=on_disk,
-                distance=distance,
-            ),
-            shard_number=self.shard_number,
-            replication_factor=self.replication_factor,
-            write_consistency_factor=self.write_consistency_factor,
-            on_disk_payload=self.on_disk_payload,
-            hnsw_config=self.hnsw_config,
-            optimizers_config=self.optimizers_config,
-            wal_config=self.wal_config,
-            quantization_config=self.quantization_config,
-            init_from=self.init_from,
-        )
+    def _recreate_collection(
+        self,
+        collection_name: str,
+        distance,
+        embedding_dim: int,
+        on_disk: bool,  # noqa: FBT001
+        use_sparse_embeddings: bool,  # noqa: FBT001
+    ):
+        if use_sparse_embeddings:
+            self.client.recreate_collection(
+                collection_name=collection_name,
+                vectors_config={
+                    DENSE_VECTORS_NAME: rest.VectorParams(
+                        size=embedding_dim,
+                        on_disk=on_disk,
+                        distance=distance,
+                    ),
+                },
+                sparse_vectors_config={
+                    SPARSE_VECTORS_NAME: rest.SparseVectorParams(
+                        index=rest.SparseIndexParams(
+                            on_disk=on_disk,
+                        )
+                    )
+                },
+                shard_number=self.shard_number,
+                replication_factor=self.replication_factor,
+                write_consistency_factor=self.write_consistency_factor,
+                on_disk_payload=self.on_disk_payload,
+                hnsw_config=self.hnsw_config,
+                optimizers_config=self.optimizers_config,
+                wal_config=self.wal_config,
+                quantization_config=self.quantization_config,
+                init_from=self.init_from,
+            )
+        if not use_sparse_embeddings:
+            self.client.recreate_collection(
+                collection_name=collection_name,
+                vectors_config=rest.VectorParams(
+                    size=embedding_dim,
+                    on_disk=on_disk,
+                    distance=distance,
+                ),
+                shard_number=self.shard_number,
+                replication_factor=self.replication_factor,
+                write_consistency_factor=self.write_consistency_factor,
+                on_disk_payload=self.on_disk_payload,
+                hnsw_config=self.hnsw_config,
+                optimizers_config=self.optimizers_config,
+                wal_config=self.wal_config,
+                quantization_config=self.quantization_config,
+                init_from=self.init_from,
+            )
```

Comment:
```suggestion
        dense_vector_name = DENSE_VECTORS_NAME if use_sparse_embeddings else ""

        vectors_config = {
            dense_vector_name: rest.VectorParams(
                size=embedding_dim,
                on_disk=on_disk,
                distance=distance,
            ),
        }

        if use_sparse_embeddings:
            vectors_config[SPARSE_VECTORS_NAME] = rest.SparseVectorParams(
                index=rest.SparseIndexParams(
                    on_disk=on_disk,
                )
            )

        self.client.recreate_collection(
            collection_name=collection_name,
            vectors_config=vectors_config,
            shard_number=self.shard_number,
            replication_factor=self.replication_factor,
            write_consistency_factor=self.write_consistency_factor,
            on_disk_payload=self.on_disk_payload,
            hnsw_config=self.hnsw_config,
            optimizers_config=self.optimizers_config,
            wal_config=self.wal_config,
            quantization_config=self.quantization_config,
            init_from=self.init_from,
        )
```

Reply:
I think this broke the test.

Reply:
@Anush008  Fixed with late commit, hope this is still not breaking change.

Reply:
This modification of code now force use to handle 3 cases:

1. Before this PR vectors and not dict just a VectorParams
2. After this PR + use sparse = True: vectors are dict with named dense "text-dense"
3. After this PR + use sprase = False: vectors are dict with named dense ""

Too many case to cover, my head will go boom.

Reply:
Feel free to pause your work. I'll look into it later

Reply:
Yikes. Sorry. `The sparse_vector_config` is a different property.

It's looking good now.

Reply:
> Yikes. Sorry. `The sparse_vector_config` is a different property.
> 
> It's looking good now.

No worries, sometimes I get frustrated while fighting with code ahah